src/connection/host_lookup.c

```diff
@@ -0,0 +1,435 @@
+	  else
+	    {
+	      strcpy (hostent_List[host_count].hostname, token);
+	      strcpy (map_hostname, token);
+	    }
+	}
+      while (token = strtok_r (NULL, delim, &save_ptr_strtok));
+//naming rule check, detailed rule, error set
+//if hostent_List[host_count].hostname is empty, set error
+      if (hostent_List[host_count].hostname != NULL)
+	{
+	  if ((user_host_Map.find (map_hostname) == user_host_Map.end ()))
+	    {
+	      user_host_Map[map_hostname] = map_ipaddr;
+	    }
+/*duplicated hostname*/
+	  else if (!strcmp (user_host_Map.find (map_hostname)->second.c_str (), hostent_List[host_count].ipaddr))
+	    {
+/*duplicated hostname and ip address both*/
+	    }
+	  else
+	    {
+/*duplicated hostname but different ip address*/
+	      return NULL;
+	    }
+	  host_count++;
+	}
+    }
+  return host_count;
+}
+
+/*
+ * gethostbyname_uhost () - Do same job with gethostbyname (), using by the 'user' defined 'hosts.conf' file.
+ * 
+ * return   : the hostent pointer.
+ * hostname (in) : the hostname.
+ *
+ */
+struct hostent *
+gethostbyname_uhost (char *name)
+{
+  struct hostent *hp;
+
+  if (host_conf_Use < 0)
+    {
+      host_conf_Use = prm_get_bool_value (PRM_ID_USE_USER_HOSTS);
+    }
+
+  if (host_conf_Use == 0)
+    {
+      return gethostbyname (name);
+    }
+
+  hp = host_lookup_internal (name, NULL, HOSTNAME_TO_IPADDR);
+
+  if (hp == NULL)
+    {
+      return NULL;
+    }
+
+  return hp;
+}
+
+//gethostbyname_r_uhost (const char *hostname, struct hostent *out_hp)
+#ifdef HAVE_GETHOSTBYNAME_R
+#if defined (HAVE_GETHOSTBYNAME_R_GLIBC)
+int
+gethostbyname_r_uhost (const char *name,
+		       struct hostent *ret, char *buf, size_t buflen, struct hostent **result, int *h_errnop)
+#elif defined (HAVE_GETHOSTBYNAME_R_SOLARIS)
+struct hostent *
+gethostbyname_r_uhost (const char *name, struct hostent *ret, char *buf, size_t buflen, int *h_errnop)
+#elif defined (HAVE_GETHOSTBYNAME_R_HOSTENT_DATA)
+int
+gethostbyname_r_uhost (const char *name, struct hostent *ret, struct hostent_data *ht_data)
+#else
+struct hostent *
+gethostbyname (const char *name)
+#error "HAVE_GETHOSTBYNAME_R"
+#endif				/* HAVE_GETHOSTBYNAME_R_GLIBC */
+#endif				/* HAVE_GETHOSTBYNAME_R */
+{
+
+  struct hostent *hp_memcpy = NULL;
+
+  if (host_conf_Use < 0)
+    {
+      host_conf_Use = prm_get_bool_value (PRM_ID_USE_USER_HOSTS);
+    }
+
+  if (host_conf_Use == 0)
+    {
+//err also modified by callee
+#ifdef HAVE_GETHOSTBYNAME_R
+#if defined (HAVE_GETHOSTBYNAME_R_GLIBC)
+      return gethostbyname_r (name, ret, buf, buflen, result, h_errnop);
+#elif defined (HAVE_GETHOSTBYNAME_R_SOLARIS)
+      return gethostbyname_r (name, ret, buf, buflen, h_errnop);
+#elif defined (HAVE_GETHOSTBYNAME_R_HOSTENT_DATA)
+      return gethostbyname_r (name, ret, &ht_data);
+#else
+      return gethostbyname (name);
+#error "HAVE_GETHOSTBYNAME_R"
+#endif
+#endif /* !HAVE_GETHOSTBYNAME_R */
+    }
+
+  hp_memcpy = host_lookup_internal (name, NULL, HOSTNAME_TO_IPADDR);
```

Comment:
How about hp instead of hp_memcpy for the variable name?

Reply:
What about **hp_buf**?

'hp' is usually used for return value itself in linux api.

But it is not used like that here.